types/firestore.d.ts

```diff
@@ -1000,6 +1000,41 @@ declare namespace FirebaseFirestore {
      * @return true if this `CollectionReference` is equal to the provided one.
      */
     isEqual(other: CollectionReference): boolean;
+
+    /**
+     * Creates and returns a new CollectionReference instance that applies a
+     * field mask and only returns the specified subset of fields for calls to
+     * `get()`, `stream()`, `getAll()` and `streamAll()`. You can specify a list
+     * of field paths to return, or use an empty list to only return the
+     * references of matching documents.
+     *
+     * This function returns a new (immutable) instance of the
+     * CollectionReference (rather than modify the existing instance) to impose
+     * the field mask.
+     *
+     * @param field The field paths to return.
+     * @return The created Query.
+     */
+    select(...field: (string | FieldPath)[]): CollectionReference;
```

Comment:
This seems odd. I would expect a `CollectionReference` to be a reference to a collection, and not also a filter on fields of documents that I might later get from that collection. If I had a vanilla `CollectionReference` and one that had been returned via `select()`, how would I tell them apart?